sdk/core/Azure.Core/src/Shared/HttpPipelineMessageHandler.cs

```diff
@@ -47,14 +66,16 @@ public static async Task<Request> ToPipelineRequestAsync(this HttpRequestMessage
             return pipelineRequest;
         }
 
-        public static HttpResponseMessage ToHttpResponseMessage(this Response response)
+        private static HttpResponseMessage ToHttpResponseMessage(Response response)
         {
             HttpResponseMessage responseMessage = new HttpResponseMessage
             {
-                StatusCode = (HttpStatusCode)response.Status,
-
-                Content = new StreamContent(response.ContentStream)
+                StatusCode = (HttpStatusCode)response.Status
             };
+            if (response.ContentStream != null)
```

Comment:
Is this a behavior change we should cover with a test?

Reply:
The test changes are what revealed this bug.